1281 replace visible to

jasonmule opened the pull request:
This pull request makes the following changes:
- Replace visible to with Published, Under review and Archived

Test these changes by:
- Checking that you can change post statuses through actions.
- Filter posts by status

Fixes ushahidi/platform#1281 .

Ping @ushahidi/platform

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/ushahidi/platform-client/290)
<!-- Reviewable:end -->

jasonmule commented:
A few things that need fixing:
- The bulk actions dropdown in the list toolbar. For some reason, this doesn't come into view in the list toolbar.
- Assigning status through bulk actions dropdown.
- Add "archived" status to the API
- Layout of the datepicker used in the post edit toolbox.

rjmackay commented:
@jasonmule do we also need API changes to enable editing a posts author? worth double-checking

rjmackay commented on app/common/directives/focus.js:
```diff
@@ -0,0 +1,20 @@
+    '$timeout'
+];
+
+function Focus($timeout) {
```
What does the focus directive do? 2 sentence doc would be great..

rjmackay commented on app/common/directives/dropdown.js:
```diff
@@ -281,6 +281,29 @@ angular.module('ushahidi.common.dropdown', ['ui.bootstrap.position'])
     };
 })
 
+.directive('isFloating', function () {
```
What does isFloating do? If this is an parameter to `dropdown` it should probably get handled in the drop down directive via `$attr`

jasonmule replied:
Should change the name of this to `unpositioned` :) This is used to position dropdown menus that are part of a parent container that's not positioned.

rjmackay commented on app/post/modify/post-editor.directive.js:
```diff
@@ -49,8 +50,10 @@ function PostEditorController(
+    PostActionsService,
+    $rootScope
   ) {
+    $rootScope.setLayout('layout-a');
```
Can you use the `layout-class` directive instead? I'm not keen on requiring $rootScope everywhere.

rjmackay commented on app/post/modify/post-editor.directive.js:
```diff
@@ -62,7 +65,6 @@ function PostEditorController(
     $scope.setVisibleStage = setVisibleStage;
     $scope.fetchAttributes = fetchAttributes;
     $scope.fetchStages = fetchStages;
-    $scope.allowedChangeStatus = allowedChangeStatus;
```
Do we not need this in the view anymore? not even to show/hide the publish control?

jasonmule replied:
I don't think we are using the visibility control here anymore?

rjmackay commented on app/post/modify/post-toolbox.directive.js:
```diff
@@ -0,0 +1,78 @@
+
+            formatDates();
+
+            $scope.changeStatus = function (status) {
```
Can you use the newer code style for this directive?
ie. named functions, bind to $scope at the top of `link`, move link to a named function.

rjmackay commented on app/post/shared/post-status.service.js:
```diff
@@ -0,0 +1,13 @@
+function (
+    Util
+) {
+    var PostStatusService = {
```
Could this just be part of the post-actions service? or post-edit service?

rjmackay commented on app/post/shared/post-actions.directive.js:
```diff
@@ -31,6 +36,20 @@ function (
+                $location.path('/posts/' + post.id + '/edit');
+            };
+
+            $scope.updateStatus = function (status) {
```
I think this is a good candidate to move into the post-actions service? or post edit service? I assume the logic is shared in a few places where we change status..

rjmackay commented on app/post/shared/post-metadata.service.js:
```diff
@@ -0,0 +1,35 @@
+module.exports = [
```
Could this file use the newer style too? https://github.com/johnpapa/angular-styleguide/blob/master/a1/README.md
Primarily use named function - so you can trace them in the debugger, and bind to the service at the top of the funciton..

rjmackay commented:
Mostly looking pretty good - but I'd like to have new code matching the angular style guide https://github.com/johnpapa/angular-styleguide/blob/master/a1/README.md
Also add some tests so that coverage at least goes up not down :)

rjmackay commented:
This still has a few outstanding comments.

Also missing API changes for:
- editable author
- editable date
I think its fine to land without those, but we need to hide the UI for both of the before it lands.

---

Reviewed 9 of 25 files at r1, 6 of 18 files at r2, 6 of 6 files at r3, 15 of 16 files at r4.
Review status: all files reviewed at latest revision, 12 unresolved discussions, some commit checks failed.

---

*[package.json, line 104 \[r4\]](https://reviewable.io:443/reviews/ushahidi/platform-client/290#-KNffW8g-xaBIP4FxWb_:-KNffW8h3dQE6h3dlhqr:544923929) ([raw file](https://github.com/ushahidi/platform-client/blob/bfd5f037e0d133fff5db2e21b8a29a3c27927db1/package.json#L104)):*
> ```JSON
>     "ngGeolocation": "rjmackay/ngGeolocation",
>     "nvd3": "^1.8.3",
>     "platform-pattern-library": "ushahidi/platform-pattern-library#gh-pages",
> ```

This should switch to `master` and we need to make sure the right changes are cherry picked/merged into `master` on the PL. That could probably be done by Seth

---

*[app/post/modify/post-editor.directive.js, line 35 \[r4\]](https://reviewable.io:443/reviews/ushahidi/platform-client/290#-KNffzUuoF5hjBl3Yxs4:-KNffzUuoF5hjBl3Yxs5:114599683) ([raw file](https://github.com/ushahidi/platform-client/blob/bfd5f037e0d133fff5db2e21b8a29a3c27927db1/app/post/modify/post-editor.directive.js#L35)):*
> ```JavaScript
>     '_',
>     'PostActionsService',
>     '$rootScope'
> ```

Remove the $rootScope dep

---

*[app/post/modify/post-toolbox.directive.js, line 52 \[r4\]](https://reviewable.io:443/reviews/ushahidi/platform-client/290#-KNfgIuSjVCcwwBKv265:-KNfgIuSjVCcwwBKv266:-1623599920) ([raw file](https://github.com/ushahidi/platform-client/blob/bfd5f037e0d133fff5db2e21b8a29a3c27927db1/app/post/modify/post-toolbox.directive.js#L52)):*
> ```JavaScript
>         }
> 
>         function allowedChangeOwner() {
> ```

I think anyone with Edit permissions should be able to change owner or timestamp - however we need API change for either of those to work.

---

*[app/post/views/filters/active-filters.directive.js, line 105 \[r4\]](https://reviewable.io:443/reviews/ushahidi/platform-client/290#-KNfghAqel5ohO9eg0H0:-KNfghAqel5ohO9eg0H1:1890069891) ([raw file](https://github.com/ushahidi/platform-client/blob/bfd5f037e0d133fff5db2e21b8a29a3c27927db1/app/post/views/filters/active-filters.directive.js#L105)):*
> ```JavaScript
>                 return $filter('date', 'longdate')(value);
>             },
>             status : function (value) {
> ```

Looks like this could all move into the view now

---

*[server/www/templates/posts/post-metadata.html, line 5 \[r4\]](https://reviewable.io:443/reviews/ushahidi/platform-client/290#-KNfhC2pFZH09u3H38Im:-KNfhC2pFZH09u3H38In:-1202872962) ([raw file](https://github.com/ushahidi/platform-client/blob/bfd5f037e0d133fff5db2e21b8a29a3c27927db1/server/www/templates/posts/post-metadata.html#L5)):*
> ```HTML
>     <span ng-show="post.user && !post.author_realname" class="metadata-author"><img src="http://www.gravatar.com/avatar/{{ post.user.gravatar || '00000000000000000000000000000000' }}?d=retro&s=40" class="avatar">{{ post.user.realname }}</span>
> 
>     <span ng-show="post.contact && !post.auther_realname" class="metadata-author">{{ post.contact.contact }}</span>
> ```

typo: auther -> author

---

*[server/www/templates/posts/post-metadata.html, line 13 \[r4\]](https://reviewable.io:443/reviews/ushahidi/platform-client/290#-KNfh9K3RBfg_F9I0MLc:-KNfh9K4l22UP4gHGWNo:1912740023) ([raw file](https://github.com/ushahidi/platform-client/blob/bfd5f037e0d133fff5db2e21b8a29a3c27927db1/server/www/templates/posts/post-metadata.html#L13)):*
> ```HTML
>     <span translate-values="{ source : source }" translate="post.source"></span>
> 
>     <!--
> ```

Why are we commenting this out? Can we just remove it? Have we replaced this with new style published/draft/archived indicator?

---

*[server/www/templates/posts/modify/post-editor.html, line 72 \[r4\]](https://reviewable.io:443/reviews/ushahidi/platform-client/290#-KNfhNJZe1x9JIcE3xgP:-KNfhNJZe1x9JIcE3xgQ:1386249432) ([raw file](https://github.com/ushahidi/platform-client/blob/bfd5f037e0d133fff5db2e21b8a29a3c27927db1/server/www/templates/posts/modify/post-editor.html#L72)):*
> ```HTML
> 
>               <!-- IF: User has permission to manage who it's visible to -->
>               <!--
> ```

Just remove it, don't comment it out.

---

*[server/www/templates/posts/views/filters/filter-posts.html, line 30 \[r4\]](https://reviewable.io:443/reviews/ushahidi/platform-client/290#-KNfhVCP3rHIeemCSU1m:-KNfhVCQXBceqwhGCPV6:1453506382) ([raw file](https://github.com/ushahidi/platform-client/blob/bfd5f037e0d133fff5db2e21b8a29a3c27927db1/server/www/templates/posts/views/filters/filter-posts.html#L30)):*
> ```HTML
> 
>             <filter-category ng-model="filters.tags"></filter-category>
>             <!--
> ```

Remove it, don't comment it out.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/ushahidi/platform-client/290#-:-KNfhnELRuPa4B1pQLRs:-1230428695)*
<!-- Sent from Reviewable.io -->

jasonmule commented:
*[app/post/shared/post-actions.directive.js, line 44 \[r2\]](https://reviewable.io:443/reviews/ushahidi/platform-client/290#-KMfIXva3NUy7rd7wNwN-r2-44:-KNfm5GpHmKL4219uNmt:-499146009) ([raw file](https://github.com/ushahidi/platform-client/blob/79de0b583f420778436c0200ca17bfc53e9ec1c2/app/post/shared/post-actions.directive.js#L44)):*
<details><summary><i>Previously, rjmackay (Robbie Mackay) wrote…</i></summary>
> I think this is a good candidate to move into the post-actions service? or post edit service? I assume the logic is shared in a few places where we change status..
</details>
Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/ushahidi/platform-client/290)*
<!-- Sent from Reviewable.io -->

jasonmule commented:
Review status: all files reviewed at latest revision, 15 unresolved discussions, some commit checks failed.

---

*[app/common/directives/dropdown.js, line 284 \[r2\]](https://reviewable.io:443/reviews/ushahidi/platform-client/290#-KMuIqWgZkHrw8fWlu1G-r2-284:-KNfp91dgZqaSj1xR7Rq:-499146009) ([raw file](https://github.com/ushahidi/platform-client/blob/79de0b583f420778436c0200ca17bfc53e9ec1c2/app/common/directives/dropdown.js#L284)):*
<details><summary><i>Previously, jasonmule (Jason Mule) wrote…</i></summary>
> Should change the name of this to `unpositioned` :) This is used to position dropdown menus that are part of a parent container that's not positioned.
</details>
Done.

---

*[app/common/directives/focus.js, line 7 \[r2\]](https://reviewable.io:443/reviews/ushahidi/platform-client/290#-KMfIXvb2gWO7ieJ9vG8-r2-7:-KNfp7IOHgsEUsgO3jEE:-499146009) ([raw file](https://github.com/ushahidi/platform-client/blob/79de0b583f420778436c0200ca17bfc53e9ec1c2/app/common/directives/focus.js#L7)):*
<details><summary><i>Previously, rjmackay (Robbie Mackay) wrote…</i></summary>
> What does the focus directive do? 2 sentence doc would be great..
</details>
Done.

---

*[app/post/modify/post-editor.directive.js, line 56 \[r2\]](https://reviewable.io:443/reviews/ushahidi/platform-client/290#-KMfIXva3NUy7rd7wNwL-r2-56:-KNfm1JenQWZR8BuzRBn:-499146009) ([raw file](https://github.com/ushahidi/platform-client/blob/79de0b583f420778436c0200ca17bfc53e9ec1c2/app/post/modify/post-editor.directive.js#L56)):*
<details><summary><i>Previously, rjmackay (Robbie Mackay) wrote…</i></summary>
> Can you use the `layout-class` directive instead? I'm not keen on requiring $rootScope everywhere.
</details>
Done.

---

*[app/post/modify/post-toolbox.directive.js, line 29 \[r2\]](https://reviewable.io:443/reviews/ushahidi/platform-client/290#-KMfIXvb2gWO7ieJ9vG9-r2-29:-KNfm3HHmJatBpgM02Ln:-499146009) ([raw file](https://github.com/ushahidi/platform-client/blob/79de0b583f420778436c0200ca17bfc53e9ec1c2/app/post/modify/post-toolbox.directive.js#L29)):*
<details><summary><i>Previously, rjmackay (Robbie Mackay) wrote…</i></summary>
> Can you use the newer code style for this directive?
> 
> ie. named functions, bind to $scope at the top of `link`, move link to a named function.
</details>
Done.

---

*[app/post/shared/post-metadata.service.js, line 1 \[r2\]](https://reviewable.io:443/reviews/ushahidi/platform-client/290#-KMfIXvb2gWO7ieJ9vGA-r2-1:-KNfmIQl80WnNeoc3M7I:-499146009) ([raw file](https://github.com/ushahidi/platform-client/blob/79de0b583f420778436c0200ca17bfc53e9ec1c2/app/post/shared/post-metadata.service.js#L1)):*
<details><summary><i>Previously, rjmackay (Robbie Mackay) wrote…</i></summary>
> Could this file use the newer style too? https://github.com/johnpapa/angular-styleguide/blob/master/a1/README.md
> 
> Primarily use named function - so you can trace them in the debugger, and bind to the service at the top of the funciton..
</details>
Done.

---

*[app/post/views/filters/active-filters.directive.js, line 105 \[r4\]](https://reviewable.io:443/reviews/ushahidi/platform-client/290#-KNfghAqel5ohO9eg0H0:-KNmGt_hDC-Islmswflx:700831696) ([raw file](https://github.com/ushahidi/platform-client/blob/bfd5f037e0d133fff5db2e21b8a29a3c27927db1/app/post/views/filters/active-filters.directive.js#L105)):*
<details><summary><i>Previously, rjmackay (Robbie Mackay) wrote…</i></summary>
> Looks like this could all move into the view now
</details>
Instead of using the transformer in the directive to process them?

---

*[app/post/shared/post-status.service.js, line 6 \[r2\]](https://reviewable.io:443/reviews/ushahidi/platform-client/290#-KMfIXvb2gWO7ieJ9vGB-r2-6:-KNfpIedKVPb0fBaONJn:-499146009) ([raw file](https://github.com/ushahidi/platform-client/blob/79de0b583f420778436c0200ca17bfc53e9ec1c2/app/post/shared/post-status.service.js#L6)):*
<details><summary><i>Previously, rjmackay (Robbie Mackay) wrote…</i></summary>
> Could this just be part of the post-actions service? or post-edit service?
</details>
Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/ushahidi/platform-client/290)*
<!-- Sent from Reviewable.io -->

jasonmule commented:
Review status: all files reviewed at latest revision, 13 unresolved discussions, some commit checks failed.

---

*[package.json, line 104 \[r4\]](https://reviewable.io:443/reviews/ushahidi/platform-client/290#-KNffW8g-xaBIP4FxWb_:-KNmHSsvEuoBnwvg1Oek:1031032105) ([raw file](https://github.com/ushahidi/platform-client/blob/bfd5f037e0d133fff5db2e21b8a29a3c27927db1/package.json#L104)):*
<details><summary><i>Previously, rjmackay (Robbie Mackay) wrote…</i></summary>
> This should switch to `master` and we need to make sure the right changes are cherry picked/merged into `master` on the PL. That could probably be done by Seth
</details>
Switched to `develop` for now.

---

*[server/www/templates/posts/post-metadata.html, line 5 \[r4\]](https://reviewable.io:443/reviews/ushahidi/platform-client/290#-KNfhC2pFZH09u3H38Im:-KNmHFukQOPlBFaf4fSt:-499146009) ([raw file](https://github.com/ushahidi/platform-client/blob/bfd5f037e0d133fff5db2e21b8a29a3c27927db1/server/www/templates/posts/post-metadata.html#L5)):*
<details><summary><i>Previously, rjmackay (Robbie Mackay) wrote…</i></summary>
> typo: auther -> author
</details>
Done.

---

*[server/www/templates/posts/post-metadata.html, line 13 \[r4\]](https://reviewable.io:443/reviews/ushahidi/platform-client/290#-KNfh9K3RBfg_F9I0MLc:-KNmHF42bBIftff4BrIj:-499146009) ([raw file](https://github.com/ushahidi/platform-client/blob/bfd5f037e0d133fff5db2e21b8a29a3c27927db1/server/www/templates/posts/post-metadata.html#L13)):*
<details><summary><i>Previously, rjmackay (Robbie Mackay) wrote…</i></summary>
> Why are we commenting this out? Can we just remove it? Have we replaced this with new style published/draft/archived indicator?
</details>
Done.

---

*[server/www/templates/posts/modify/post-editor.html, line 72 \[r4\]](https://reviewable.io:443/reviews/ushahidi/platform-client/290#-KNfhNJZe1x9JIcE3xgP:-KNmHGlwJgsGCF2vpaJu:-499146009) ([raw file](https://github.com/ushahidi/platform-client/blob/bfd5f037e0d133fff5db2e21b8a29a3c27927db1/server/www/templates/posts/modify/post-editor.html#L72)):*
<details><summary><i>Previously, rjmackay (Robbie Mackay) wrote…</i></summary>
> Just remove it, don't comment it out.
</details>
Done.

---

*[server/www/templates/posts/views/filters/filter-posts.html, line 30 \[r4\]](https://reviewable.io:443/reviews/ushahidi/platform-client/290#-KNfhVCP3rHIeemCSU1m:-KNmHOQgClPBh7ZP2kIx:-499146009) ([raw file](https://github.com/ushahidi/platform-client/blob/bfd5f037e0d133fff5db2e21b8a29a3c27927db1/server/www/templates/posts/views/filters/filter-posts.html#L30)):*
<details><summary><i>Previously, rjmackay (Robbie Mackay) wrote…</i></summary>
> Remove it, don't comment it out.
</details>
Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/ushahidi/platform-client/290)*
<!-- Sent from Reviewable.io -->

rjmackay commented:
Reviewed 6 of 11 files at r5.
Review status: 30 of 35 files reviewed at latest revision, 6 unresolved discussions, some commit checks failed.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/ushahidi/platform-client/290)*
<!-- Sent from Reviewable.io -->
